snap/component.go

```diff
@@ -61,6 +62,13 @@ func (ci *ComponentInfo) Provenance() string {
 	return ci.ComponentProvenance
 }
 
+func (ci *ComponentInfo) Version(snapVersion string) string {
```

Comment:
Hm, we already store some things from `snap.Info` inside of `snap.ComponentInfo` (see the `Hooks` field). Would this be made better by having a `snapVersion` field, and then this `Version` method could take no parameters? Potentially returning an error if `snapVersion` and `YamlVersion` are both unset?

Definitely not for sure a better solution, just something to consider.

Reply:
This might help reduce some of the changes in `seedwriter`, I think.

Reply:
I see, that makes sense. Although, to use `comp.Version(snapVersion)` in those cases, we'd need the `snap.Info` anyways, right?

Reply:
Yes, that's right but we need to use the component version in very few places fortunately.